### PR TITLE
Add test cases to test_xen.py

### PR DESCRIPTION
**Pull request by hkx303**
**Description**
- Positive
- [x] test_encrypted_password
- [x] test_hypervisor_id
- [x] test_filter_hosts
- [x] test_exclude_hosts
- [x] test_fake_type
- Negative
- [x] test_type
- [x] test_server
- [x] test_username
- [x] test_password
- [x] test_encrypted_password
- [x] test_filter_hosts
- [x] test_exclude_host

### CODE REVIEW

File: tests/hypervisor/test_xen.py [hunk 1]
```diff
@@ -7,42 +7,659 @@
...
+        )
+        result = virtwho.run_service()
+        assert (
+            result["error"] is not 0
+            and result["send"] == 1
+            and result["thread"] == 1
+            and assertion["disable_multi_configs"] in result["error_msg"]
+        )
+
+        # username option is null but another config is ok
```
Comment (Junefen) [inline tests/hypervisor/test_xen.py:404]: could delete this steps

File: tests/hypervisor/test_xen.py [hunk 2]
```diff
@@ -7,42 +7,659 @@
...
+        )
+        result = virtwho.run_service()
+        assert (
+            result["error"] is not 0
+            and result["send"] == 1
+            and result["thread"] == 1
+            and assertion["disable_multi_configs"] in result["error_msg"]
+        )
+
+        # password option is null but another config is ok
```
Comment (Junefen) [inline tests/hypervisor/test_xen.py:472]: also could delete this step

File: tests/hypervisor/test_xen.py [hunk 3]
```diff
@@ -7,42 +7,659 @@
...
+        function_hypervisor.delete("type")
+        result = virtwho.run_service()
+        assert (
+            result["error"] is not 0
+            and result["send"] == 0
+            and result["thread"] == 1
+            and assertion["disable"] in result["error_msg"]
+        )
+
+        # type option is disable but another config is ok
```
[the remark this answers is not in the document]
Comment (hkx303) [reply tests/hypervisor/test_xen.py]: Removed

File: tests/hypervisor/test_xen.py (hunk 1 above)
Comment (hkx303) [reply tests/hypervisor/test_xen.py]: Removed

File: tests/hypervisor/test_xen.py (hunk 2 above)
Comment (hkx303) [reply tests/hypervisor/test_xen.py]: Removed

**Review by Junefen — APPROVED** [review]
ACK

**Pull request merged**